### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by chernokulskyi**
[DEMO LINK](https://antch8.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,34 +1,28 @@
...
+export const App: React.FC = () => {
+  const [movies, setMovies] = useState<Movie[]>(moviesFromServer);
 
-export class App extends React.Component<{}, State> {
-  state: State = {
-    movies: moviesFromServer,
+  const addMovie = (movie: Movie) => {
+    setMovies((prevMovies) => ([
+      ...prevMovies, movie,
+    ]));
```
[the remark this answers is not in the document]
Comment (jstmpelowycz) [reply src/App.tsx:14]: In case you pass `setMovies` directly you could avoid using `useCallback`

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,16 +1,135 @@
...
+}
 
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState<string>('');
+  const [description, setDescription] = useState<string>('');
+  const [imgUrl, setImgUrl] = useState<string>('');
+  const [imdbUrl, setImdbUrl] = useState<string>('');
+  const [imdbId, setImdbId] = useState<string>('');
```
Comment (jstmpelowycz) [inline src/components/NewMovie/NewMovie.tsx:13]: No need to insert generic types here

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,16 +1,135 @@
...
+            name="title"
+            placeholder="title"
+            value={title}
+            onChange={changeHandler}
+            autoComplete="off"
+            className="input"
+            required
+          />
+        </div>
+      </div>
```
Comment (jstmpelowycz) [inline src/components/NewMovie/NewMovie.tsx:73]: Looks like a separate component

**Review by jstmpelowycz — COMMENTED** [review]
A few things to consider, good job though 😏🚀

**Review by jstmpelowycz — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ihor-Karpyn) [reply src/App.tsx:14]: Agree, let's add useCallback

**Review by Ihor-Karpyn — APPROVED** [review]

**Pull request closed without merging**